src/main/scala/memeid/Constructors.scala

```diff
@@ -37,4 +44,31 @@ trait Constructors {
       })
     }
 
+  def v3[F[_]: Sync, A](namespace: UUID, local: A)(implicit D: Digestible[A]): F[UUID] =
+    Sync[F].delay {
+
+      val digest = MD5.digest
```

Comment:
Both `v3` and `v5` constructors could be refactored with a method like:

```scala
def hashedUUID[A: Digestible](algorithm: Algorithm, version: Short, namespace: UUID, name: A)
```

Reply:
Done!